code/datums/components/rust.dm

```diff
@@ -0,0 +1,83 @@
+/datum/component/rust
```

Comment:
A LOT of things use rust, I'm worried about memory usage. You should make this an element, as it doesn't seem to carry any state other than...overlay? Which I bet we can grab at deletion time.

Reply:
The component is actually very small memory wise, and I need to keep a reference to the MA and not just create a new one each time the signal fires for the overlay |= operator